packages/graphiql/src/components/HeaderEditor.tsx

```diff
@@ -0,0 +1,234 @@
+/**
+ *  Copyright (c) 2019 GraphQL Contributors.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ */
+import * as CM from 'codemirror';
+import 'codemirror/addon/hint/show-hint';
+import React from 'react';
+
+import onHasCompletion from '../utility/onHasCompletion';
+import commonKeys from '../utility/commonKeys';
+
+declare module CodeMirror {
+  export interface Editor extends CM.Editor {}
+  export interface ShowHintOptions {
+    completeSingle: boolean;
+    hint: CM.HintFunction | CM.AsyncHintFunction;
+    container: HTMLElement | null;
+  }
+}
+
+type HeaderEditorProps = {
+  value?: string;
+  onEdit: (value: string) => void;
+  readOnly?: boolean;
+  onHintInformationRender: (value: HTMLDivElement) => void;
+  onPrettifyQuery: (value?: string) => void;
+  onMergeQuery: (value?: string) => void;
+  onRunQuery: (value?: string) => void;
+  editorTheme?: string;
+};
+
+/**
+ * HeaderEditor
+ *
+ * An instance of CodeMirror for editing headers to be passed with the GraphQL request.
+ *
+ * Props:
+ *
+ *   - value: The text of the editor.
+ *   - onEdit: A function called when the editor changes, given the edited text.
+ *   - readOnly: Turns the editor to read-only mode.
+ *
+ */
+export class HeaderEditor extends React.Component<HeaderEditorProps> {
+  CodeMirror: any;
+  editor: (CM.Editor & { options: any }) | null = null;
+  cachedValue: string;
+  private _node: HTMLElement | null = null;
+  ignoreChangeEvent: boolean;
+  constructor(props: HeaderEditorProps) {
+    super(props);
+
+    // Keep a cached version of the value, this cache will be updated when the
+    // editor is updated, which can later be used to protect the editor from
+    // unnecessary updates during the update lifecycle.
+    this.cachedValue = props.value || '';
+    this.ignoreChangeEvent = true;
```

Comment:
Initialize `ignoreChangeEvent` to false. Else it will not be triggered on the first render

Reply:
[This](https://github.com/graphql/graphiql/commit/e54e1a8691483f1d336231314130d9822481b3be) is the recent fix in Query Variables